src/codeCache.cpp

```diff
@@ -159,13 +159,17 @@ const void* CodeCache::findSymbolByPrefix(const char* prefix) {
 }
 
 const void* CodeCache::findSymbolByPrefix(const char* prefix, int prefix_len) {
+    const void* result = NULL;
     for (int i = 0; i < _count; i++) {
         const char* blob_name = _blobs[i]._name;
         if (blob_name != NULL && strncmp(blob_name, prefix, prefix_len) == 0) {
-            return _blobs[i]._start;
+            result = _blobs[i]._start;
+            if (strchr(blob_name + prefix_len, '.') == NULL) {
```

Comment:
Add a short comment describing what we are doing here

Reply:
4fbb4e1dd15c3a4ccfb26e60469aea1abdda6534